Content.Server/Atmos/Components/FilterMaskComponent.cs

```diff
@@ -1,16 +1,14 @@
-///using Content.Shared.Inventory;
-using Content.Server.Body.Systems;
-namespace Content.Server.Atmos.Components
+
+namespace Content.Server.Atmos.Components;
+
+/// <summary>
+/// Used in head and mask with a smoke gas filter.
+/// </summary>
+[RegisterComponent]
+[ComponentProtoName("FilterMask")]
```

Comment:
replace these 2 lines with `[RegisterComponent, Access(typeof(SmokeFilterSystem))]`

Reply:
Done, do you still want i use event system ?

Reply:
it would be ideal for maintainability but itll still work without it, see what maints think